src/transformStateWithClones.js

```diff
@@ -7,7 +7,37 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const cloneObject = {};
+  const arrayWithObjects = [];
+
+  Object.assign(cloneObject, state);
+
+  for (const objects of actions) {
+    const { type, extraData, keysToRemove } = objects;
```

Comment:
You want your code to be maximally clear for those who read it and for yourself too.
```suggestion
  for (const action of actions) {
    const { type, extraData, keysToRemove } = action;
```